Make domain block/silence/reject-media code more robust

ClearlyClaire opened the pull request:
In the past years, we have received multiple reports of domain blocks/silence/suspension not being carried out completely/reliably. I think most, if not all, of these issues have been fixed for new blocks in 3.1 (thanks to the rework of domain blocks in 3.0, and the fix in #12247).

However, this PR proposes the following additional changes:
- Split media cleanup from reject-media domain blocks to its own service:
  This makes the code somewhat easier to read and separates failure to delete media in the reject-media case from failure to mark accounts as silenced or suspended.
  This also allows moving the deletion of existing media to a lower queue.
- Slightly improve handling of some missing files in the reject-media case. I'm not sure why this code would be needed, and haven't experienced such issues myself, but there is no reason to think they'd occur in the `BackupService` (#13146) and not the deletion of remote media.
- Ensure Rails cache for every affected status is cleared, even if something errored out in the way
- Do not catch `ActiveRecord::RecordNotFound` in the worker, as it might possibly catch actual errors, not just the `DomainBlock` object not existing anymore.
- Immediately mark every account of blocked domain as suspended, rather than doing so sequentially, one by one, all the while deleting their data. Clearing the data isn't sped up, but the actual block is.

ClearlyClaire committed:
Split media cleanup from reject-media domain blocks to its own service

ClearlyClaire committed:
Slightly improve ClearDomainMediaService error handling

ClearlyClaire committed:
Lower DomainClearMediaWorker to lowest-priority queue

ClearlyClaire committed:
Do not catch ActiveRecord::RecordNotFound in domain block workers

ClearlyClaire committed:
Fix DomainBlockWorker spec labels

ClearlyClaire committed:
Add some specs

Gargron commented on app/workers/domain_block_worker.rb:
```diff
@@ -4,8 +4,11 @@ class DomainBlockWorker
-  rescue ActiveRecord::RecordNotFound
-    true
+    domain_block = DomainBlock.find_by(id: domain_block_id)
+    if domain_block.nil?
```
What's the benefit of rewriting it this way?

ThibG replied:
The `ActiveRecord::RecordNotFound` error we want to catch is the one coming from the `DomainBlock` not existing, not one that could arise from a bug in `BlockDomainService`.

Gargron replied:
In that case I would recommend using a guard clause, the return value `true` has no impact (in the rescue, it's just a way to avoid the don't supress exceptions code style check), to keep nesting level down.

ClearlyClaire committed:
Change domain blocks to immediately mark accounts as suspended

Rather than doing so sequentially, account after account, while cleaning
their data. This doesn't change much about the time the block takes to
complete, but it immediately prevents interaction with the blocked domain,
while up to now, it would only be guaranteed when the process ends.
